internal/provider/provider.go

```diff
@@ -406,6 +406,12 @@ func New() *schema.Provider {
 						ForceNew:    true,
 						Optional:    true,
 					},
+					"icon": {
+						Type:        schema.TypeString,
+						Description: "The Material UI icon name in CamelCase.",
```

Comment:
Instead of letting people pick any icon we should just provide a list of icon types such as `["compute", "storage", "unknown"]` etc. and map those to icons. This way we avoid locking ourselves into material UI icons

Reply:
Hm... I think this is reasonable but kinda limited. No strong opinions tho. cc: @bpmct

Reply:
Hmm... kinda sucks that we can't do the entire set but I understand why as we might want to change icon sets down the road.

Thoughts on also supporting arbitrary icon URLs, the same way the `coder_app` resource does? This shouldn't necessarily be blocking, but ideally these things should be as customizable as possible as long as there aren't technical limitations... For example, in templates:

- use any resource
- use any name
- use any parameters
- add any app, with any icon URL
- add any metadata
- add any instructions in README
- ideally: use any icon for resources

Reply:
In the interest of keeping things simple since this is not a very important feature IMO unlike app icons, we should do the URL support in the future

Reply:
I think we should only do icon URLs, not `material-ui` icons. We don't use the latest `material-ui` version, so it might be difficult for users to understand what they can or cannot use.

The relative path has been working well for us, so I'd say we go down that route. Thoughts @bpmct ?

Reply:
I do like using a common "image" mechanism for templates, apps, and resources.  Perhaps we could download some svg/material icons and serve them statically?